tests/test_ext_marshmallow.py

```diff
@@ -515,48 +515,54 @@ def test_schema_uses_ref_if_available_v3(self, spec_fixture):
             "schema"
         ] == build_ref(spec_fixture.spec, "schema", "Pet")
 
-    def test_schema_uses_ref_if_available_name_resolver_returns_none_v2(self):
+    @pytest.mark.parametrize(
+        "pet_schema", (PetSchema, PetSchema(), "tests.schemas.PetSchema"),
+    )
+    def test_schema_name_resolver_returns_none_v2(self, pet_schema):
```

Comment:
These tests were previously not testing a unique code path. Changed them to expose this bug. We could add back the old tests as well but I'm not sure they were adding any value.